auditor/src/client/mod.rs

```diff
@@ -323,6 +457,21 @@ impl AuditorClient {
             .json()
             .await?)
     }
+
+    #[tracing::instrument(
+        name = "Getting custom record queries from AUDITOR server.",
+        skip(self)
+    )]
+    pub async fn advanced_query(&self, query_string: String) -> Result<Vec<Record>, ClientError> {
```

Comment:
Documentation: Please add some documentation to the function (like you did for the blocking client). See also my comment there.